### PR TITLE
tf: various improvements

### PR DESCRIPTION
**Pull request by howardjohn**

### CODE REVIEW

File: tests/integration/README.md [hunk 1]
```diff
@@ -497,7 +497,7 @@ The framework accepts standard istio logging flags. You can use these flags to e
 framework, as well as some of the components that are used in-line in the native environment:
 
 ```console
-$ go test ./... --log_output_level=tf:debug,mcp:debug
+$ go test ./... --log_output_level=tf:debug
```
Comment (howardjohn) [inline tests/integration/README.md:500]: mcp is no longer there

File: tests/integration/pilot/common/apps.go [hunk 2]
```diff
@@ -209,7 +209,7 @@ func SetupApps(t resource.Context, i istio.Instance, apps *EchoDeployments) erro
 			Subsets:        []echo.SubsetConfig{{}},
 		})
 
-	skipDelta := t.Settings().Skip(echo.Delta) || !t.Settings().Revisions.AtLeast("1.11")
+	skipDelta := t.Settings().Skip(echo.Delta) || !t.Settings().Revisions.AtLeast("1.12")
```
Comment (howardjohn) [inline tests/integration/pilot/common/apps.go:212]: delta was not working until 1.12

File: tests/integration/pilot/multicluster_test.go [hunk 3]
```diff
@@ -209,6 +209,39 @@ func TestBadRemoteSecret(t *testing.T) {
 
 				t.ConfigKube().YAML(secret).ApplyOrFail(t, ns)
 			}
+			// Test exec auth
+			// CreateRemoteSecret can never generate this, so create it manually
```
Comment (howardjohn) [inline tests/integration/pilot/multicluster_test.go:213]: extend some test coverage

File: tests/integration/security/main_test.go [hunk 4]
```diff
@@ -39,7 +40,7 @@ func TestMain(m *testing.M) {
 		NewSuite(m).
 		Setup(istio.Setup(&ist, setupConfig)).
 		Setup(func(ctx resource.Context) error {
-			return util.SetupApps(ctx, ist, apps, true)
+			return util.SetupApps(ctx, ist, apps, !ctx.Settings().Skip(echo.VM))
```
Comment (howardjohn) [inline tests/integration/security/main_test.go:43]: do not deploy VM when we skip VM

**Review by nmittler — APPROVED** [review]

**Review by lei-tang — APPROVED** [review]

**Pull request merged**